### DISCUSSION TITLE
How do I get the cursor position?

### DISCUSSION THREAD

**Discussion by patsornk** [ueberdosis/tiptap#1764] [General Questions] [answered]
At this question. Can I get the cursor position or selected-node.

**Comment by philippkuehn** [comment, marked as answer]
you should check `editor.state.selection`

https://prosemirror.net/docs/ref/#state.Selection

**Reply by patsornk** [reply]
I don't understand how to use it.

**Reply by patsornk** [reply]
If I want to set focus at next line. Can I ?

**Reply by Arik14** [reply]
_Focus at the next line_ is another question.
But for cursor position, try playing with event handler onSelectionUpdate, use console.log, just like mine.
![Screenshot from 2021-09-01 15-50-22](https://user-images.githubusercontent.com/54788602/131642768-21fe2310-b841-4e0a-a997-9c0ca4d3eb38.png)

Using that, when I select this text,
![Screenshot from 2021-09-01 15-55-36](https://user-images.githubusercontent.com/54788602/131642890-a1ff27b7-e102-4f84-a708-ce11ac4966ce.png)
we could see the console. Now expand the Editor in the console, you will this nested object.
Find state, then selection, you will find range, from, to. These are the cursor position.
![Screenshot from 2021-09-01 16-02-39](https://user-images.githubusercontent.com/54788602/131643693-897b376b-89d3-4b81-8bc2-d207f40cb694.png)

But in order to understand which position value _from_ & _to_ refers to, you need to understand how position and selection works in ProseMirror (which I'm not fluent with it). Good luck.
